nixos/modules/services/misc/gitlab.nix

```diff
@@ -105,22 +119,22 @@ let
     GITLAB_UPLOADS_PATH = "${cfg.statePath}/uploads";
     GITLAB_LOG_PATH = "${cfg.statePath}/log";
     GITLAB_SHELL_PATH = "${cfg.packages.gitlab-shell}";
-    GITLAB_SHELL_CONFIG_PATH = "${cfg.statePath}/shell/config.yml";
+    GITLAB_SHELL_CONFIG_PATH = "${cfg.statePath}/home/config.yml";
```

Comment:
I don't see that, in `bin/compile` they move all the binaries to that same directory from `$BUILD_DIR/bin`. It gets its configuration from the [current directory](https://gitlab.com/gitlab-org/gitlab-shell/blob/v5.1.1/go/internal/config/config.go#L30) though, so it seems it assumes it will run from the root of `gitlab-shell` directory.

A much better approach would be to either to patch that code or change their running directories.

If they are going to be moved, it would be best to keep `shell` in their name to make it clear what component of the Gitlab architecture they are part of.